src/newTSFeature/codeGenerator.ts

```diff
@@ -0,0 +1,100 @@
+import { existsSync, writeFile } from "fs";
+import path = require("path");
+import { getTSDataLayerRepositoryCodeTemplate } from "../code_generator_templates/ts/ts_data_layer_template";
+import { getTSDomainLayerCodeTemplate } from "../code_generator_templates/ts/ts_domain_layer_template";
+import { getTSDeserializerCodeTemplate, getTSOrchestratorCodeTemplate, getTSSerializerCodeTemplate } from "../code_generator_templates/ts/ts_gateway_layer_template";
+import { TSFeatureType } from "../utils/enums";
+
+export const generateTSCode = (
+
+    featureClassName: string,
+	snakeCaseFeatureName: string,
+	targetDirectory: string,
+    type: TSFeatureType,
+
+
+): void => {
+
+
+	if (type == TSFeatureType.NEW_OPERATION) {
+
+        const targetPath = `${targetDirectory}/${snakeCaseFeatureName}`
+
+
+		generateDataLayerCode(`${targetPath}/data/repository`, featureClassName, snakeCaseFeatureName);
+
+		generateDomainLayerCode(`${targetPath}/domain/repository/`, featureClassName,snakeCaseFeatureName);
+
+	} else {
+
+        const targetPath = `${targetDirectory}/gateway/${snakeCaseFeatureName}`
+
+        generateGateWayLayerCode(targetPath, featureClassName, snakeCaseFeatureName);
+	}	
+
+}
+
+function generateDataLayerCode(
+	targetDirectory: string,
+	featureClassName: string,
+	snakeCaseFeatureName: string,
+) {
+	writeDataToFile(
+		targetDirectory,
+		`${snakeCaseFeatureName}RepositoryImpl.ts`,
+		getTSDataLayerRepositoryCodeTemplate(featureClassName, snakeCaseFeatureName)
+	);
+}
+
+
+
+function generateDomainLayerCode(
+	targetDirectory: string,
+	featureClassName: string,
+    snakeCaseFeatureName: string
+) {
+	writeDataToFile(
+		targetDirectory,
+		`${snakeCaseFeatureName}Repository.ts`,
+		getTSDomainLayerCodeTemplate(featureClassName)
+	);
+}
+
+function generateGateWayLayerCode(
+	targetDirectory: string,
+	featureClassName: string,
+	snakeCaseFeatureName: string,
+) { 
+	writeDataToFile(
+		targetDirectory,
+		`index.ts`,
+		getTSOrchestratorCodeTemplate(featureClassName, snakeCaseFeatureName)
+	);
+
+	writeDataToFile(
+		targetDirectory,
+		`${snakeCaseFeatureName}RequestDeSerializer.ts`,
+		getTSDeserializerCodeTemplate(featureClassName)
+	);
+
+	writeDataToFile(
+		targetDirectory,
+		`${snakeCaseFeatureName}RequestSerializer.ts`,
+		getTSSerializerCodeTemplate(featureClassName)
+	);
+	
+}
+
+
+function writeDataToFile(targetDirectory: string, fileName: string, dataToWrite: string) {
+	const filePath = path.join(`${targetDirectory}`, fileName)
+	if (existsSync(filePath)) {
+		throw Error(`${fileName} already exists`);
+	}
+
+	writeFile(
+		filePath,
+		dataToWrite,
+		function () { },
+	);
+}
```

Comment:
This can be moved to utils folder right?

It is used in both places

Reply:
Cool
will do that